internal/webhooks/patch_validation.go

```diff
@@ -198,6 +199,36 @@ func validateSelectors(selector clusterv1.PatchSelector, class *clusterv1.Cluste
 	if selector.MatchResources.MachineDeploymentClass != nil && len(selector.MatchResources.MachineDeploymentClass.Names) > 0 {
 		for i, name := range selector.MatchResources.MachineDeploymentClass.Names {
 			match := false
+			if strings.Contains(name, "*") {
+				// selector can at most have a single * rune
+				if strings.Count(name, "*") > 1 {
+					allErrs = append(allErrs, field.Invalid(
+						path.Child("matchResources", "machineDeploymentClass", "names").Index(i),
+						name,
+						"selector can at most contain a single \"*\" rune"))
+					break
+				}
+
+				// the * rune can appear only at the beginning, or ending of the selector.
+				if strings.Contains(name, "*") && !(strings.HasPrefix(name, "*") || strings.HasSuffix(name, "*")) {
+					// templateMDClass can only have "*" rune at the start or end of the string
+					allErrs = append(allErrs, field.Invalid(
+						path.Child("matchResources", "machineDeploymentClass", "names").Index(i),
+						name,
+						"\"*\" rune can only appear at the beginning, or ending of the selector"))
+					break
+				}
+				// a valid selector with/out "*" should comply with Kubernetes naming standards.
+				if validation.IsQualifiedName(strings.ReplaceAll(name, "*", "a")) != nil {
+					allErrs = append(allErrs, field.Invalid(
+						path.Child("matchResources", "machineDeploymentClass", "names").Index(i),
+						name,
+						"selector does not comply with the Kubernetes naming standards"))
+					break
+				}
+				match = true
+				break
+			}
 			for _, md := range class.Spec.Workers.MachineDeployments {
 				if md.Class == name {
 					if selectorMatchTemplate(selector, md.Template.Infrastructure.Ref) ||
```

Comment:
Can you please modify:
```
            names:
            - default-worker
```

to
```
            names:
            - *-worker
```
in test/e2e/data/infrastructure-docker/v1beta1/main/clusterclass-quick-start.yaml

Just gives us some e2e test coverage that the ClusterClass webhook allows the `*` syntax in selectors

Reply:
>in test/e2e/data/infrastructure-docker/v1beta1/main/clusterclass-quick-start.yaml

This file doesn't exist anymore. Can you point me to an alternate one?

Reply:
Now test/e2e/data/infrastructure-docker/main/clusterclass-quick-start.yaml